### ISSUE TITLE
Add new device to HWDB (for libinput consumption)

### ISSUE THREAD

**Issue by arigit** [systemd/systemd#3947]
@whot 
### Submission type
- [ ] Bug report
- [ X] Request for enhancement (RFE)

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

> 229 (Fedora 24)

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_
### Used distribution

> Fedora 24

As discussed with  whot on #wayland (freenode) 

This request is to add the Click Angle for a new device to the **hwdb** :

**Device:**  
Input device name: "Logitech MX Master"
Input device ID: bus 0x03 vendor 0x46d product 0x4041 version 0x111

Main Vertical scroll wheel: 24 clicks for a full revolution
    Reported by evemu-record as (for each click):
          E: 9.273909 0002 0008 -001    # EV_REL / **REL_WHEEL**  
Horizontal scroll wheel AKA "thumbwheel":  generates 14 events for a full revolution (it's a tiny wheel)
    Reported by evemu-record as (for each 'click'):
          E: 11.859882 0002 0006 0001   # EV_REL / **REL_HWHEEL** 

Notes: 
The main wheel supports "free spinning mode" or "ratchet mode". It produces 24 events in either mode.
The smaller 'horizontal scroll wheel' is always in free spinning mode (no ratchet) but spins slowly (no inertia)

The bottom line request is to have equivalent functionality in libinput as we had with evdev, where we could add "dividers" to the events generated by either wheel, for example:

xinput set-prop 11 "Evdev Scrolling Distance" 1 2 1

... this caused the Horizontal Wheel to send to the app a single horizontal scroll keystroke for every 2 "mouse whell events", in effect decreasing the "sensitivity" of the wheel.

**Comment by whot** [comment]
fwiw, this is a special case where the hwheel has a different click angle than the normal wheel. we'll need to add a new tag/format for this, the current udev property assumes that both wheels are identical

**Comment by whot** [comment]
I'll need to you once again check on the 14 events for a revolution value. This gives us an angle of 25.7 per click which seems odd. Then again the hw designers may have just decided that's the number of clicks we need. Please make sure it's definitely not 15 clicks per rotation.

**Comment by arigit** [comment]
@whot : I fully understand. Re-tested **many** times with the horizontal scrollwheel, it's definitely not 15, it's 14. Tested around 30 times in each direction (up, and down). Please note that this wheel is way smaller than the normal vertical scroll wheel and is mounted on the side (so it can be rotated "up" or "down"). Also, this wheel has no "ratchet" = it is constantly in "free moving" mode and this makes the testing a bit complicated. Sometimes I get 13 events particularly when moving "down" but most of the time is 14. I think I can conclude with reasonable certainty that it is 14. Notice also that I _never_ got 15 events. 

I also asked ppl with the MX mouse in the solaar thread (referenced above) to share results. I do notice it is not many of us using libinput just yet.

**Comment by whot** [comment]
sorry, had to force-push a few times, didn't realise this would cause comment spam :)

@arigit this is the systemd bit and the libinput bit is ready to push as well. if you have an account in the fdo bugzilla please file a bug against Input/libinput in the xorg component. the actual conversion from click angle to click events is done by the xf86-input-libinput driver.

**Comment by arigit** [comment]
@whot  this is done: https://bugs.freedesktop.org/show_bug.cgi?id=97404

**Comment by whot** [comment]
for the archives, commit a6a8e60bd318 changed this to MOUSE_WHEEL_CLICK_ANGLE_HORIZONTAL
